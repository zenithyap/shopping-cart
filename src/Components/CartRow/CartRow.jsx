import styles from './cartrow.module.css';

export default function CartRow({ item, handleAddClick, handleRemoveClick }) {
  return (
    <div className={styles.row}>
      <div className={styles.imageTitle}>
        <div className={styles.imageContainer}>
          <img src={item.image} alt={item.title} />
        </div>
        <p>{item.title}</p>
      </div>
      <div className={styles.spacer}></div>
      <p>Total Price: ${item.price * item.quantity}</p>
      <div>
        <p>Quantity: {item.quantity}</p>
        <button className={styles.leftButton} onClick={() => handleRemoveClick(item)}>-</button>
        <button onClick={() => handleAddClick(item)}>+</button>
      </div>
    </div>
  );
}
