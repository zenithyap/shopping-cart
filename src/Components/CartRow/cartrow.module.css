.row {
    display: flex;
    height: 200px;
    justify-content: space-around;
    border: 2px solid rgb(196, 196, 196);
    border-radius: 20px;
    align-items: center;
    text-align: center;
    padding: 1rem;
    gap: 1rem;
}

.row * {
    flex: 1;
}

.imageContainer {
    height: 150px;
}

.imageContainer img {
    width: 100%;
    height: 100%;
    object-fit: contain;
}

.row button {
    padding: 5px 10px;
    width: 35px;
    height: 35px;
}

.buttonsContainer {
    display: flex;
    gap: 0.5rem;
    width: fit-content;
    margin: 0 auto;
}

@media (max-width: 600px) {
    .row p {
        font-size: 0.8rem;
    }

    .buttonsContainer {
        gap: 0.2rem;
    }
}