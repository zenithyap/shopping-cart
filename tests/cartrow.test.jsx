import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import CartRow from '../src/Components/CartRow/CartRow';

describe('CartRow component', () => {
  it('Should display values properly', () => {
    const mockItem = {
      title: 'Mock Item',
      price: 2,
      quantity: 2,
    };

    const { container } = render(<CartRow item={mockItem} />);

    expect(container).toMatchSnapshot();
  });

  it('Handles user interactions correctly', async () => {
    const mockHandleAddClick = vi.fn();
    const mockHandleRemoveClick = vi.fn();

    const mockItem = {
      title: 'Mock Item',
      price: 2,
      quantity: 2,
    };

    const user = userEvent.setup();

    render(
      <CartRow
        item={mockItem}
        handleAddClick={mockHandleAddClick}
        handleRemoveClick={mockHandleRemoveClick}
      />,
    );

    await user.click(screen.getByRole('button', {name: '+'}));
    expect(mockHandleAddClick).toHaveBeenCalled();

    await user.click(screen.getByRole('button', {name: '-'}));
    expect(mockHandleRemoveClick).toHaveBeenCalled();
  });
});
