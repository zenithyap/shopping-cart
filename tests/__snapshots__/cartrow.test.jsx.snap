// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CartRow component > Should display values properly 1`] = `
<div>
  <div
    class="_row_611b28"
  >
    <div
      class="_imageTitle_611b28"
    >
      <div
        class="_imageContainer_611b28"
      >
        <img
          alt="Mock Item"
        />
      </div>
      <p>
        Mock Item
      </p>
    </div>
    <div
      class="_spacer_611b28"
    />
    <p>
      Total Price: $
      4
    </p>
    <div>
      <p>
        Quantity: 
        2
      </p>
      <button
        class="_leftButton_611b28"
      >
        -
      </button>
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;
